contracts/mainnet/connectors/liquity/main.sol

```diff
@@ -0,0 +1,347 @@
+     * @param setWithdrawId Optional storage slot to store the withdrawn ETH to
+     * @param getRepayId Optional storage slot to retrieve the LUSD to repay
+     * @param setBorrowId Optional storage slot to store the LUSD borrowed
+    */
+    function adjust(
+        uint maxFeePercentage,
+        uint withdrawAmount,
+        uint depositAmount,
+        uint borrowAmount,
+        uint repayAmount,
+        address upperHint,
+        address lowerHint,
+        uint getDepositId,
+        uint setWithdrawId,
+        uint getRepayId,
+        uint setBorrowId
+    ) external payable returns (string memory _eventName, bytes memory _eventParam) {
+        AdjustTrove memory adjustTrove;
+
+        adjustTrove.maxFeePercentage = maxFeePercentage;
+        adjustTrove.withdrawAmount = withdrawAmount;
+        adjustTrove.depositAmount = getUint(getDepositId, depositAmount);
+        adjustTrove.borrowAmount = borrowAmount;
+        adjustTrove.repayAmount = getUint(getRepayId, repayAmount);
+        adjustTrove.isBorrow = borrowAmount > 0;
+
+        borrowerOperations.adjustTrove{value: depositAmount}(
+            adjustTrove.maxFeePercentage,
+            adjustTrove.withdrawAmount,
+            adjustTrove.borrowAmount,
+            adjustTrove.isBorrow,
+            upperHint,
+            lowerHint
+        );
+        
+        // Allow other spells to use the withdrawn collateral
+        setUint(setWithdrawId, withdrawAmount);
+
+        // Allow other spells to use the borrowed amount
+        setUint(setBorrowId, borrowAmount);
+
+        _eventName = "LogAdjust(address,uint,uint,uint,uint,uint,uint,uint,uint,uint)";
+        _eventParam = abi.encode(msg.sender, maxFeePercentage, depositAmount, borrowAmount, getDepositId, setWithdrawId, getRepayId, setBorrowId);
+    }
+
+    /**
+     * @dev Withdraw remaining ETH balance from user's redeemed Trove to their DSA
+     * @notice Claim remaining collateral from Trove
+    */
+    function claimCollateralFromRedemption() external returns(string memory _eventName, bytes memory _eventParam) {
+        borrowerOperations.claimCollateral();
+        _eventName = "LogClaimCollateralFromRedemption(address)";
+        _eventParam = abi.encode(msg.sender);
+    }
+    /* End: Trove */
+
+    /* Begin: Stability Pool */
+
+    /**
+     * @dev Deposit LUSD into Stability Pool
+     * @notice Deposit LUSD into Stability Pool
+     * @param amount Amount of LUSD to deposit into Stability Pool
+     * @param frontendTag Address of the frontend to make this deposit against (determines the kickback rate of rewards)
+     * @param getId Optional storage slot to retrieve the LUSD from
+    */
+    function stabilityDeposit(
+        uint amount,
+        address frontendTag,
+        uint getId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
+        amount = getUint(getId, amount);
+
+        stabilityPool.provideToSP(amount, frontendTag);
```

Comment:
In all operations it would be nice  to have `setUint`’s for LQTY and ETH gains.

Reply:
https://github.com/liquity/dsa-connectors/pull/1/commits/97db94fe595330743385f26c30e15ab28cc18063